core/src/OpenACC/Kokkos_OpenACC_ParallelFor_MDRange.hpp

```diff
@@ -0,0 +1,243 @@
+    auto const end4   = m_policy.m_upper[4];
+    auto const begin3 = m_policy.m_lower[3];
+    auto const end3   = m_policy.m_upper[3];
+    auto const begin2 = m_policy.m_lower[2];
+    auto const end2   = m_policy.m_upper[2];
+    auto const begin1 = m_policy.m_lower[1];
+    auto const end1   = m_policy.m_upper[1];
+    auto const begin0 = m_policy.m_lower[0];
+    auto const end0   = m_policy.m_upper[0];
+
+    if ((end0 <= begin0) || (end1 <= begin1) || (end2 <= begin2) ||
+        (end3 <= begin3) || (end4 <= begin4)) {
+      return;
+    }
+
+    // avoid implicit capture of *this which would yield a memory access
+    // violation when executing on the device
+    auto const& functor(m_functor);
+
+    int const async_arg = m_policy.space().acc_async_queue();
+
+#pragma acc parallel loop gang vector collapse(5) copyin(functor) \
+    async(async_arg)
+    for (auto i4 = begin4; i4 < end4; ++i4) {
+      for (auto i3 = begin3; i3 < end3; ++i3) {
+        for (auto i2 = begin2; i2 < end2; ++i2) {
+          for (auto i1 = begin1; i1 < end1; ++i1) {
+            for (auto i0 = begin0; i0 < end0; ++i0) {
+              functor(i0, i1, i2, i3, i4);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  template <int Rank>
+  std::enable_if_t<Rank == 6> execute_impl() const {
+    auto const begin5 = m_policy.m_lower[5];
+    auto const end5   = m_policy.m_upper[5];
+    auto const begin4 = m_policy.m_lower[4];
+    auto const end4   = m_policy.m_upper[4];
+    auto const begin3 = m_policy.m_lower[3];
+    auto const end3   = m_policy.m_upper[3];
+    auto const begin2 = m_policy.m_lower[2];
+    auto const end2   = m_policy.m_upper[2];
+    auto const begin1 = m_policy.m_lower[1];
+    auto const end1   = m_policy.m_upper[1];
+    auto const begin0 = m_policy.m_lower[0];
+    auto const end0   = m_policy.m_upper[0];
+
+    if ((end0 <= begin0) || (end1 <= begin1) || (end2 <= begin2) ||
+        (end3 <= begin3) || (end4 <= begin4) || (end5 <= begin5)) {
+      return;
+    }
+
+    // avoid implicit capture of *this which would yield a memory access
+    // violation when executing on the device
+    auto const& functor(m_functor);
+
+    int const async_arg = m_policy.space().acc_async_queue();
+
+#pragma acc parallel loop gang vector collapse(6) copyin(functor) \
+    async(async_arg)
+    for (auto i5 = begin5; i5 < end5; ++i5) {
+      for (auto i4 = begin4; i4 < end4; ++i4) {
+        for (auto i3 = begin3; i3 < end3; ++i3) {
+          for (auto i2 = begin2; i2 < end2; ++i2) {
+            for (auto i1 = begin1; i1 < end1; ++i1) {
+              for (auto i0 = begin0; i0 < end0; ++i0) {
+                functor(i0, i1, i2, i3, i4, i5);
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  void execute() const {
+    static_assert(Policy::rank < 7,
+                  "OpenACC Backend MDRangePolicy Error: Unsupported rank...");
+    execute_impl<Policy::rank>();
+  }
+};
```

Comment:
Please say that only 1<ranks<7 are supported. Rank=1 should trigger the static assert too.

Reply:
Updated.